tools/map/fetch_modern_admin_boundaries.py

```
@@ -0,0 +1,283 @@
+    if record_index != len(rows):
+        raise ValueError("boundary SHP/DBF record count mismatch")
+    return features
+
+
+def fetch_boundary_archive(recipe: BoundaryRecipe, output: Path | str) -> BoundaryArtifact:
+    destination = Path(output)
+    destination.parent.mkdir(parents=True, exist_ok=True)
+    temporary = destination.with_suffix(destination.suffix + ".part")
+    try:
+        request = urllib.request.Request(recipe.url, headers={"User-Agent": "OpenSamguk-map-builder/1"})
+        with urllib.request.urlopen(request) as response:
```

Ruff (0.16.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)